### PR TITLE
correcting resource(string) and using the openshift override

### PR DESCRIPTION
**Pull request by shawkins**
## Description
fix for #3329 and #3330 - related to the resource(String/HasMetadata) operations

With a little more effort a test could be added for the openshift readiness when using a resource method.

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.java [hunk 1]
```diff
@@ -104,7 +101,7 @@ public NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl(OkHttpCl
     this(client, config, namespace, explicitNamespace, fromServer, deletingExisting, visitors, unmarshal(is), -1, null, cascading, watchRetryInitialBackoffMillis, watchRetryBackoffMultiplier, dryRun);
   }
 
-  public NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl(OkHttpClient client, Config config, String namespace, String explicitNamespace, Boolean fromServer, Boolean deletingExisting, List<Visitor> visitors, Object item, long gracePeriodSeconds, DeletionPropagation propagationPolicy, Boolean cascading, long watchRetryInitialBackoffMillis, double watchRetryBackoffMultiplier, boolean dryRun) {
+  public NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl(OkHttpClient client, Config config, String namespace, String explicitNamespace, Boolean fromServer, Boolean deletingExisting, List<Visitor> visitors, HasMetadata item, long gracePeriodSeconds, DeletionPropagation propagationPolicy, Boolean cascading, long watchRetryInitialBackoffMillis, double watchRetryBackoffMultiplier, boolean dryRun) {
```
Comment (metacosm) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.java:104]: Are we sure this doesn't have side-effects?

Comment (manusa) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.java:104]: This part is tightly related to #3329 

This constructor is specifically used here:
https://github.com/fabric8io/kubernetes-client/blob/69c6fcdce3745a153931a6c45f3a7c44941959b9/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BaseKubernetesClient.java#L208-L209
and:
https://github.com/fabric8io/kubernetes-client/blob/69c6fcdce3745a153931a6c45f3a7c44941959b9/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BaseKubernetesClient.java#L216-L217

Which previously was:
https://github.com/fabric8io/kubernetes-client/blob/c24152096f1a0438be1cd4978bea23a8dc185334/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BaseKubernetesClient.java#L215-L216

So considering that #33229 is a bug and the String actually represents a serialized/marshalled resource:
https://github.com/fabric8io/kubernetes-client/blob/1f5061f1f04b046ed0fb40b2a9a7432b72b452cb/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/KubernetesClient.java#L348-L355

This shouldn't have any side-effect and actually fix an existing issue/bug.

**Review by manusa — APPROVED** [review]
Awesome, thank you very much!

**Review by rohanKanojia — APPROVED** [review]

**Comment by rohanKanojia** [comment]
Sonar seems to be reporting about the usage of `DEFAULT_INITIAL_BACKOFF_MILLIS`, `DEFAULT_BACKOFF_MULTIPLIER`. Why are we using deprecated constants? Javadoc says they are no longer being used:
https://github.com/fabric8io/kubernetes-client/blob/921a469bd7d69c41f08c0a9486bab7deccc8d4e2/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Waitable.java#L23-L27

But I'm still able to see some usages of these across the project.

**Comment by shawkins** [comment]
> But I'm still able to see some usages of these across the project.

I deprecated them, but removing looked to a major under taking.  I'm fine with getting rid of them, but at this point I'd wait until the other major change for the resourcehandlers is committed.

**Review by metacosm — APPROVED** [review]

**Pull request merged**